qcoro/qcorotask.h

```diff
@@ -732,4 +733,35 @@ inline T waitFor(QCoro::Task<T> &&task) {
     return detail::waitFor<T>(task);
 }
 
+//! Connect a callback to be called when the asynchronous task finishes.
+/*!
+ * Allows to register a callback to be called only when the context object
+ * still exists when the task finishes.
+ *
+ * In contrast to the then function, the result of connect can not be co_awaited.
+ *
+ * @param context A QObject that still needs to exist when the task finishes in order for the callback to be invoked.
+ * @param func The function that will be called whe the task finishes.
+ *        For void tasks, it needs to have no arguments.
+ *        For all other types, it takes a value of the type as single argument.
+ */
+template <typename T, typename Callback>
+requires std::is_invocable_v<Callback> || std::is_invocable_v<Callback, T>
+void connect(QCoro::Task<T> &task, QObject *context, Callback func) {
+    QPointer ctxWatcher = context;
+    if constexpr (std::is_same_v<T, void>) {
+        task.then([=, func = std::move(func)]() {
+            if (ctxWatcher) {
+                func();
+            }
+        });
+    } else {
+        task.then([=, func = std::move(func)](auto &&value) {
+            if (ctxWatcher) {
+                func(std::move(value));
```

Comment:
I think it's recommended to do `std::forward<decltype(value)>(value)` to get move semantics where appropriate and avoid unnecessary copying.